builtin/logical/pki/path_acme_revoke.go

```diff
@@ -136,14 +136,14 @@ func (b *backend) acmeRevocationHandler(acmeCtx *acmeContext, _ *logical.Request
 func (b *backend) acmeRevocationByPoP(acmeCtx *acmeContext, userCtx *jwsCtx, cert *x509.Certificate, config *crlConfig) (*logical.Response, error) {
 	// Since this account does not exist, ensure we've gotten a private key
 	// matching the certificate's public key.
-	signer, ok := userCtx.Key.Key.(crypto.Signer)
+	public, ok := userCtx.Key.Key.(crypto.PublicKey)
 	if !ok {
 		return nil, fmt.Errorf("unable to revoke certificate: unable to parse JWS key of type (%T): %w", userCtx.Key.Key, ErrMalformed)
 	}
 
 	// Ensure that our PoP is indeed valid.
```

Comment:
comment above could say something about we have proven possession of the key in the certificate; "public" is a really hard variable to use here - and more specific (proofKey or certKey?) would be nicer - and which is which on the error message looks confusing.